### PR TITLE
Transformed most functions into async functions

### PR DESCRIPTION
**Pull request by JavierLeon9966**

### CODE REVIEW

File: src/KygekTeam/KygekEatHeal/EatHeal.php [hunk 1]
```diff
@@ -81,68 +81,81 @@ private function getHealValue(Player $player) : float {
...
+            return;
+        }
 
         $price = (int) $this->getConfig()->getNested("price.heal", 0);
         if ($this->economyEnabled && $isPlayer && $price > 0) {
             $name = $senderPlayer !== null ? $senderPlayer->getName() : $player->getName();
-            if (($result = $this->processTransaction($name, $price)) !== null) {
-                return $result;
-            }
+            $this->processTransaction($name, $price),
```
Comment (KygekDev) [inline src/KygekTeam/KygekEatHeal/EatHeal.php:120]: Syntax error

Comment (JavierLeon9966) [reply src/KygekTeam/KygekEatHeal/EatHeal.php]: Fixed

**Review by KygekDev — CHANGES_REQUESTED** [review]
There are about 20 lints, please fix them if possible:
https://poggit.pmmp.io/ci/KygekTeam/KygekEatHeal/~/pr:2

**Comment by JavierLeon9966** [comment]
Done

**Comment by KygekDev** [comment]
@JavierLeon9966 Let me know if you want me to try fix the issues instead.

**Comment by KygekDev** [comment]
This guy isn't responding, I think he's busy with his personal life and his server. I can't wait much longer, so I guess I'm going to try fix the issues tomorrow (in my time zone). Hopefully I can get the plugin updated on Poggit ASAP!

NB: If JavierLeon9966 reads this comment, I hope he can give a response.

**Comment by JavierLeon9966** [comment]
Sorry for not responding, you mean the issues at the issue tracker?

**Comment by KygekDev** [comment]
> Sorry for not responding, you mean the issues at the issue tracker? Or the one I told you about the `int|string` type hint? Because I already fixed that

The issues I meant are the problems that needs to be addressed in this issue, including the last review I made a week ago.

**Comment by JavierLeon9966** [comment]
There are no lints in the latest build ([here](https://poggit.pmmp.io/ci/KygekTeam/KygekEatHeal/~/pr:7))

**Comment by KygekDev** [comment]
> There are no lints in the latest build ([here](https://poggit.pmmp.io/ci/KygekTeam/KygekEatHeal/~/pr:7))

Okay, but could you read and comment to the review I started above?

This one:
image

**Comment by JavierLeon9966** [comment]
You didn't send that yet, as you can see is still in pending

**Review by KygekDev — CHANGES_REQUESTED** [review]
Please fix the issue below.

File: src/KygekTeam/KygekEatHeal/EatHeal.php [hunk 2]
```diff
@@ -81,68 +81,81 @@ private function getHealValue(Player $player) : float {
...
+                    $healValue = $this->getHealValue($player);
+                    $player->setHealth($healValue);
+                    $callback($price);
+                }
+            );
         }
-
-        $healValue = $this->getHealValue($player);
-        $player->setHealth($healValue);
-        return $price;
```
[the remark this answers is not in the document]
Comment (KygekDev) [reply src/KygekTeam/KygekEatHeal/EatHeal.php:116]: @JavierLeon9966 it's here now, thanks for letting me know.

[the remark this answers is not in the document]
Comment (JavierLeon9966) [reply src/KygekTeam/KygekEatHeal/EatHeal.php:116]: You are right, I forgot about that part. I'm gonna fix it soon

[the remark this answers is not in the document]
Comment (KygekDev) [reply src/KygekTeam/KygekEatHeal/EatHeal.php:116]: Thanks, I'm waiting.

[the remark this answers is not in the document]
Comment (JavierLeon9966) [reply src/KygekTeam/KygekEatHeal/EatHeal.php:116]: Fixed

**Comment by JavierLeon9966** [comment]
Is everything done now?

**Comment by KygekDev** [comment]
> Is everything done now?

I haven't tested the plugin, but I'll try to do it this week, especially because I'm no longer very busy.

**Review by AIPTU — APPROVED** [review]

**Comment by KygekDev** [comment]
I haven't tested it yet, so hopefully it works...

**Comment by MintoD** [comment]
> I haven't tested it yet, so hopefully it works...

Don't worry, I merged it and then I will check codes, next I'll upgrade it

**Pull request merged**